src/currentWeather/SearchPanel.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+import "./SearchPanel.css";
+import { SearchHistory } from "./SearchHistory";
+
+type SearchPanelProps = {
+  isActive: boolean;
+  onSetIsActive: React.Dispatch<React.SetStateAction<boolean>>;
+  onSubmit: (location: string) => Promise<void>;
+  searchHistory: string[];
+};
+
+export const SearchPanel = ({
+  isActive,
+  onSetIsActive,
+  onSubmit,
+  searchHistory,
+}: SearchPanelProps) => {
+  const [location, onSetLocation] = useState("");
+
+  function handleSubmit(event: React.FormEvent<HTMLElement>) {
+    event.preventDefault();
+
+    if (location.trim() === "") {
+      alert("Please enter a valid location.");
+      return;
+    }
+
+    onSubmit(location);
+    onSetIsActive(false);
```

Comment:
The same, do this stuff in parent, this component doesn't work with activity state or location